rules.mk

```
@@ -39,23 +39,27 @@ endif
 # parses tools.go and returns the tool name prefixed with bin/
 toolsBins := $(addprefix bin/,$(notdir $(shell grep '^\s*_' tools.go | awk -F'"' '{print $$2}')))
 
-# installs cli tools defined in tools.go
-$(toolsBins): go.sum tools.go
-	go install $$(awk -F'"' '/$(@F)/{print $$2}' tools.go)
-	
+# build cli tools defined in tools.go
+$(toolsBins): go.mod go.sum tools.go
+$(toolsBins): CMD=$(shell awk -F'"' '/$(@F)"/ {print $$2}' tools.go)
+$(toolsBins):
+	go install "$(CMD)"
+
 generated_go_files := \
 	client/cacher/mock_cacher/cacher_mock.go \
 	client/tinkerbell/mock_workflow/workflow_mock.go \
 	client/tinkerbell/mock_hardware/hardware_mock.go \
 	syslog/facility_string.go \
 	syslog/severity_string.go \
-	
-.PHONY: $(generated_go_files)
 
 # go generate
 go_generate:
 $(filter %_string.go,$(generated_go_files)): bin/stringer
 $(filter %_mock.go,$(generated_go_files)): bin/mockgen
+packet/mock_cacher/cacher_mock.go: packet/models_cacher.go
+packet/mock_workflow/workflow_mock.go: packet/models_tinkerbell.go
```

These paths need updating
```suggestion
client/cacher/mock_cacher/cacher_mock.go: client/cacher/discovery.go
client/tinkerbell/mock_workflow/workflow_mock.go: client/tinkerbell/discovery.go
client/tinkerbell/mock_hardware/hardware_mock.go: client/tinkerbell/discovery.go
```

---

fixed, good catch.